tensorpipe/common/tensor.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#if TENSORPIPE_HAS_CUDA_IPC_CHANNEL // TODO: Generic TENSORPIPE_HAS_CUDA.
+#include <cuda_runtime.h>
+#endif // TENSORPIPE_HAS_CUDA_IPC_CHANNEL
+
+namespace tensorpipe {
+
+enum class DeviceType {
+  kCpu = 0,
+#if TENSORPIPE_HAS_CUDA_IPC_CHANNEL
+  kCuda,
+#endif // TENSORPIPE_HAS_CUDA_IPC_CHANNEL
+};
+
+struct CpuTensor {
+  void* data{nullptr};
```

Comment:
Mega-nit: in some cases we called this `ptr`. Not sure whether I prefer that or `data`. Maybe `ptr` conveys more the idea that this struct doesn't hold the data by itself (it doesn't store a `string`, a `vector`, ... which owns the data) but is just a shallow copyable reference to some data that exists elsewhere...

Reply:
Yes, I kept the name as it was in `Message`, but I'd go for `ptr` as well.